-------------------------------------------------------------------------------
Test set: org.apache.hadoop.ozone.scm.TestGetCommittedBlockLengthAndPutKey
-------------------------------------------------------------------------------
Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 29.96 s <<< FAILURE! - in org.apache.hadoop.ozone.scm.TestGetCommittedBlockLengthAndPutKey
tesPutKeyResposne  Time elapsed: 0.197 s  <<< ERROR!
java.io.IOException: 
Failed to execute command cmdType: PutBlock
containerID: 3
datanodeUuid: "e9b44407-90bb-4880-a735-210111230e3b"
putBlock {
  blockData {
    blockID {
      containerID: 3
      localID: 1617257633877
      blockCommitSequenceId: 0
    }
    chunks {
      chunkName: "1617257633877.data.0"
      offset: 0
      len: 0
      checksumData {
        type: CRC32
        bytesPerChecksum: 1048576
      }
    }
    size: 0
  }
}

	at org.apache.hadoop.hdds.scm.XceiverClientSpi.getIOExceptionForSendCommand(XceiverClientSpi.java:152)
	at org.apache.hadoop.hdds.scm.XceiverClientSpi.sendCommand(XceiverClientSpi.java:118)
	at org.apache.hadoop.ozone.scm.TestGetCommittedBlockLengthAndPutKey.tesPutKeyResposne(TestGetCommittedBlockLengthAndPutKey.java:174)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.ExecutionException: org.apache.ratis.protocol.exceptions.StateMachineException: org.apache.hadoop.hdds.scm.container.common.helpers.StorageContainerException: Chunk file not found: /github/workspace/mnt/ozone/hadoop-ozone/integration-test/target/test-dir/MiniOzoneClusterImpl-a90b6438-3a8b-4d97-86d9-923ea11a648b/datanode-2/data-0/containers/hdds/a90b6438-3a8b-4d97-86d9-923ea11a648b/current/containerDir0/3/chunks/1617257633877.block
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1908)
	at org.apache.hadoop.hdds.scm.XceiverClientSpi.sendCommand(XceiverClientSpi.java:112)
	... 13 more
Caused by: org.apache.ratis.protocol.exceptions.StateMachineException: org.apache.hadoop.hdds.scm.container.common.helpers.StorageContainerException: Chunk file not found: /github/workspace/mnt/ozone/hadoop-ozone/integration-test/target/test-dir/MiniOzoneClusterImpl-a90b6438-3a8b-4d97-86d9-923ea11a648b/datanode-2/data-0/containers/hdds/a90b6438-3a8b-4d97-86d9-923ea11a648b/current/containerDir0/3/chunks/1617257633877.block
	at org.apache.hadoop.ozone.container.common.transport.server.ratis.ContainerStateMachine.lambda$applyTransaction$7(ContainerStateMachine.java:785)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1609)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more

